ui/components/src/context/PersesThemeProvider.tsx

```diff
@@ -0,0 +1,195 @@
+      fontFamily: primaryFontFamily,
+      fontSize: 12,
+    },
+    grid: {
+      top: 5,
+      right: 20,
+      bottom: 0,
+      left: 20,
+      containLabel: true,
+    },
+    color: ['#8dd3c7', '#bebada', '#fb8072', '#80b1d3', '#fdb462'],
+    categoryAxis: {
+      show: true,
+      axisLabel: {
+        show: true,
+        color: primaryTextColor,
+        margin: 15,
+      },
+      axisTick: {
+        show: false,
+        length: 6,
+        lineStyle: {
+          color: mdGrey,
+        },
+      },
+      axisLine: {
+        show: true,
+        lineStyle: {
+          color: mdGrey,
+        },
+      },
+      splitLine: {
+        show: true,
+        lineStyle: {
+          width: 0.5,
+          color: ltGrey,
+          opacity: 0.6,
+        },
+      },
+      splitArea: {
+        show: false,
+        areaStyle: {
+          color: [ltGrey],
+        },
+      },
+    },
+    valueAxis: {
+      show: true,
+      axisLabel: {
+        color: primaryTextColor,
+        margin: 12,
+      },
+      axisLine: {
+        show: false,
+      },
+      splitLine: {
+        show: true,
+        lineStyle: {
+          width: 0.5,
+          color: ltGrey,
+          opacity: 0.6,
+        },
+      },
+    },
+    legend: {
+      show: false,
+      type: 'scroll',
+      bottom: 0,
+      textStyle: {
+        color: primaryTextColor,
+      },
+      pageTextStyle: {
+        color: mdGrey,
+      },
+      pageIconColor: muiTheme?.palette?.action?.active,
+      pageIconInactiveColor: muiTheme?.palette?.action?.disabled,
+    },
+    toolbox: {
+      show: true,
+      top: 10,
+      right: 10,
+      iconStyle: {
+        borderColor: primaryTextColor,
+      },
+    },
+    tooltip: {},
+    line: {
+      showSymbol: false,
+      symbol: 'circle',
+      symbolSize: 4,
+      smooth: false,
+      lineStyle: {
+        width: 1,
+      },
+      emphasis: {
+        lineStyle: {
+          width: 1.5,
+        },
+      },
+    },
+    bar: {
+      barMaxWidth: 150,
+      itemStyle: {
+        borderWidth: 0,
+        borderColor: ltGrey,
+      },
+    },
+    gauge: {
+      detail: {
+        fontSize: 18,
+        fontWeight: 600,
+        valueAnimation: false,
+      },
+      splitLine: {
+        splitNumber: 2,
+        distance: 0,
+        length: 4,
+        lineStyle: {
+          width: 1,
+        },
+      },
+    },
+  };
+
+  const persesTheme: PersesChartsTheme = {
+    themeName,
+    theme: merge(echartsTheme, themeOverrides),
+    noDataOption: {
+      title: {
+        show: true,
+        textStyle: {
+          color: primaryTextColor,
+          fontSize: 16,
+          fontWeight: 400,
+        },
+        text: 'No data',
+        left: 'center',
+        top: 'center',
+      },
+      xAxis: {
+        show: false,
+      },
+      yAxis: {
+        show: false,
+      },
+    },
+  };
+
+  return (
+    <ChartsThemeProvider themeName={themeName} persesTheme={persesTheme}>
```

Comment:
Do we still need to pass in `themeName` if it's already available in `persesTheme`?

Reply:
Looking at it again, the way I named the two providers is confusing--I'll try to think of better names and add a comment with more info 🤔 

The reason I went with two providers is so the ChartsThemeProvider could be used independently from PersesThemeProvider. This would allow apps to not have to necessarily take on the @mui/material dependency, they could just pass an ECharts theme in directly instead of utilizing the MUI theme -> ECharts theme mapping that happens in PersesThemeProvider.

Reply:
Discussed this with Luke offline and it will be possible to refactor so there is only 1 ChartsThemeProvider, I'll let you / @Nexucis know when this and the other comments are fixed (hopefully sometime tomorrow)